### PR TITLE
Remove @Flaky annotation and disable testNotAnalzyed for MySql 8

### PR DESCRIPTION
**Pull request by leveyja**
## Description
```
    Remove @Flaky annotation and disable testNotAnalzyed for MySql 8

    See https://dev.mysql.com/doc/refman/8.0/en/innodb-parameters.html#sysvar_innodb_stats_auto_recalc
    MySql automatically analyzes tables under certain circumstances, so testNotAnalyzed is not
    deterministic
```

```
 Use randomTableSuffix() in BaseTestMySqlTableStatisticsTest tests

When increasing @test(invocationCount = X) to reproduce @flaky tests,
having a unique table name is important for test isolation.
```

> Is this change a fix, improvement, new feature, refactoring, or other?

Test fix

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Test

> How would you describe this change to a non-technical end user or system administrator?

Skip a test which cannot be run deterministically due to MySql8's stats guarantees.

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

Fixes https://github.com/trinodb/trino/issues/12413

### CODE REVIEW

File: plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/BaseTestMySqlTableStatisticsTest.java [hunk 1]
```diff
@@ -81,13 +81,9 @@ protected QueryRunner createQueryRunner()
...
     @Override
     @Test
-    @Flaky(
-            // TODO replace @Flaky with assertEventually or @Flaky-like annotation for same purpose
-            issue = "TODO",
-            match = "Expecting.*to be close to|ComparisonFailure.*NDV for")
     public void testNotAnalyzed()
     {
-        String tableName = "test_not_analyzed";
+        String tableName = "test_not_analyzed_" + randomTableSuffix();
```
Comment (wendigo) [inline plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/BaseTestMySqlTableStatisticsTest.java:86]: Nit: extract randomized table names to separate commit

**Review by wendigo — APPROVED** [review]

File: plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/BaseTestMySqlTableStatisticsTest.java (hunk 1 above)
Comment (hashhar) [reply plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/BaseTestMySqlTableStatisticsTest.java:86]: And add reasoning in commit message + PR description since it won't be obvious to other people who don't have the context.

Comment (leveyja) [reply plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/BaseTestMySqlTableStatisticsTest.java:86]: 👍

Comment (leveyja) [reply plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/BaseTestMySqlTableStatisticsTest.java:86]: @hashhar - see updated PR Title + Description (separated into 2 commits - 2nd commit removes @Flaky + disables `testNotAnalyzed` - 1st commit is preparatory in case any tests fail in future / we can re-run them with unique table names, etc - let me know if the description / commit messages could do with some work - thank you!

**Review by wendigo — APPROVED** [review]

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
Merged, thanks!
